torch/csrc/jit/ir/ir.h

```diff
@@ -1342,6 +1342,28 @@ struct ProfileOp : public Node {
   std::function<void(std::vector<IValue>&)> callback_;
 };
 
+struct TORCH_API ProfileOptionalOp : public Node {
+  static constexpr Symbol Kind = ::c10::prim::profile_optional;
```

Comment:
It's a little unfortunate that this is basically a copy-pasta of the prim::profile node... I tried inheriting from it but ran into sum difficulties with the class Kind and some other APIs. Maybe in a follow up we can try to fix the inheritance and make the final prim::profile op that doesnt have any inputs its own op symbol. I think this is fine for now.